src/test/java/com/spotify/sparkey/extra/ThreadLocalWithFallbackTest.java

```diff
@@ -0,0 +1,77 @@
+package com.spotify.sparkey.extra;
+
+import org.junit.Test;
+
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class ThreadLocalWithFallbackTest {
+
+  private final AtomicInteger closedThreads = new AtomicInteger();
+  private final ThreadLocalWithFallback<Long> threadLocal =
+          ThreadLocalWithFallback.withInitial(() -> Thread.currentThread().getId(), threadId -> closedThreads.incrementAndGet());
+
+  @Test
+  public void testSimple() {
+    assertEquals(1, threadLocal.get().intValue());
+  }
+
+  @Test
+  public void testSet() {
+    assertEquals(Thread.currentThread().getId(), threadLocal.get().longValue());
+    threadLocal.set(123L);
+    assertEquals(123, threadLocal.get().longValue());
+    threadLocal.remove();
+    assertEquals(Thread.currentThread().getId(), threadLocal.get().longValue());
+  }
+
+  @Test
+  public void testForkJoinPoolCommonPool() {
+    final int poolSize = ForkJoinPool.commonPool().getParallelism();
+    final AtomicInteger failures = new AtomicInteger();
+    for (int i = 0; i < 10000; i++) {
+      int finalI = i;
+      ForkJoinPool.commonPool().execute(() -> {
+        checkFailure(failures);
+        if (finalI < poolSize) {
+          try {
+            Thread.sleep(1);
+          } catch (InterruptedException e) {
+          }
+        }
+      });
+    }
+    assertEquals(poolSize, threadLocal.fallbackSize());
```

Comment:
couldn't it be potentially `poolSize + 1` because commonpool can end up using the main thread for optimization? Wondering as this test might fail sometimes because of that

Reply:
I guess the test should be rewritten if that happens, but not needed right now

Reply:
I added a check for main thread